### ISSUE TITLE
Feature: Support returning multiple collections from query

### ISSUE THREAD

**Issue by mfbx9da4** [Nozbe/WatermelonDB#1399]
It would be great if queries could return multiple collections!

**Comment by KrisLau** [comment]
You can do it with raw SQL queries look at `Q.unsafeSqlQuery`

**Comment by mfbx9da4** [comment]
It can’t return multiple types though

**Comment by KrisLau** [comment]
Not sure what you mean by that

**Comment by mfbx9da4** [comment]
Suppose you have a collection of posts and a collection of authors. I would like to return posts and authors from one query

**Comment by KrisLau** [comment]
Yeah you can do that with raw SQL queries. Anything you can do in SQL you can do with that

**Comment by mfbx9da4** [comment]
No not really, you have to manually construct the models and understand the inner workings of the model caches. It's not supported out of the box. You can return raw json objects but not models of different kinds.

**Comment by PEZO19** [comment]
@mfbx9da4 May I ask if you have figured out a workaround/approach?

**Comment by mfbx9da4** [comment]
Yes I inspected how the underlying driver works and built a solution using it. I don't have the code around though. A better workaround is to do multiple queries though if it makes sense for your use case as you don't need to patch watermelon.

**Comment by PEZO19** [comment]
@mfbx9da4 Thanks a lot!
